lib/BoundaryClickWatcher/index.js

```diff
@@ -76,21 +76,24 @@ class BoundaryClickWatcher extends Component {
   }
 
   handleClick(event) {
-    const targetIsContainer = event.target !== this.container;
+    const { ignoreClass } = this.props;
+    const { target } = event;
+
+    const targetIsContainer = target !== this.container;
     let containerDoesNotContainTarget = false;
     let containerDoesNotContainFocus = false;
-    let targetHasIgnoreClass = false;
     if (this.container) {
-      containerDoesNotContainTarget = !this.container.contains(event.target);
+      containerDoesNotContainTarget = !this.container.contains(target);
       containerDoesNotContainFocus = !this.container.contains(
         document.activeElement
       );
     }
-    if (this.props.ignoreClass) {
-      targetHasIgnoreClass = event.target.classList.contains(
-        this.props.ignoreClass
-      );
-    }
+
+    const ignoreClasses = ignoreClass ? ignoreClass.match(/\S+/g) : [];
```

Comment:
i dont really mind either way, is doing this better than `.split(' ')`? or is it a much of a muchness

Reply:
I originally went with split `.split('  ')`, but that doesn't allow for the fact that someone might but a class string with two blank spaces, or a tab.

This covers all the whitespace options, as it matches anything that _isn't_ whitespace.

That said, this is exactly the same as `ignoreClass.split(/\s+/g)`, which splits by any white space. I don't think there's any advantage or disadvantage to this over what is currently used.